main.tf

```diff
@@ -1,16 +1,84 @@
+locals {
+  security_group_id = var.create_before_destroy ? aws_security_group.sg_lifecycle[0].id : aws_security_group.sg_no_lifecycle[0].id
+  ingress_rules = flatten([
+    for rule in var.ingress_rules : [
+      for port in rule.ports : [
+        for source in rule.sources : [
+          for protocol in rule.protocols : {
+            description = try(rule.description, null)
+            port        = port
+            protocol    = protocol
+            source      = source
+          }
+        ]
+      ]
+    ]
+  ])
+  egress_rules = flatten([
+    for rule in var.egress_rules : [
+      for port in rule.ports : [
+        for source in rule.sources : [
+          for protocol in rule.protocols : {
+            description = try(rule.description, null)
+            port        = port
+            protocol    = protocol
+            source      = source
+          }
+        ]
+      ]
+    ]
+  ])
+}
 
-module "tags" {
-  source  = "rhythmictech/tags/terraform"
-  version = "~> 1.1.0"
+resource "aws_security_group" "sg_lifecycle" {
+  count       = var.create_before_destroy ? 1 : 0
+  name_prefix = var.name_prefix
+  description = var.description
+  tags        = var.tags
+  vpc_id      = var.vpc_id
 
-  enforce_case = "UPPER"
-  names        = [var.name]
-  tags         = var.tags
+  lifecycle {
+    create_before_destroy = true
+  }
 }
 
-locals {
-  # tflint-ignore: terraform_unused_declarations
-  name = module.tags.name
-  # tflint-ignore: terraform_unused_declarations
-  tags = module.tags.tags_no_name
+resource "aws_security_group" "sg_no_lifecycle" {
+  count       = var.create_before_destroy ? 0 : 1
+  name_prefix = var.name_prefix
```

Comment:
using name prefix is a pain when you're dealing with pets. Idk, I'd give users the option